src/test/tree/DelegationManagerUnit.tree

```diff
@@ -0,0 +1,310 @@
+│       │   └── it should push a stake update
+│       ├── it should increment the staker's cumulative withdrawals queueWithdrawals
+│       ├── it should calculate and set the withdrawal root as pending
+│       └── it should emit a WithdrawalQueued event and return the withdrawal root
+├── completeQueuedWithdrawal ***
+│   ├── given that the exiting the withdrawal queue is paused
+│   │   └── it should revert
+│   ├── given that the function is reentered
+│   │   └── it should revert
+│   └── it should call _completeQueuedWithdrawal (internal function)
+├── completeQueuedWithdrawals
+│   ├── given that the exiting the withdrawal queue is paused
+│   │   └── it should revert
+│   ├── given that the function is reentered
+│   │   └── it should revert
+│   └── it should loop through each strategy and call _completeQueuedWithdrawal (internal function)
+├── when _completeQueuedWithdrawal is called
+│   ├── given that the withdrawal root is not pending
+│   │   └── it should revert
+│   ├── given that the withdrawal delay blocks period has not elapsed
+│   │   └── it should revert
+│   ├── given that the caller is not the withdrawer
+│   │   └── it should revert
+│   ├── given that receiveAsTokens is true
+│   │   └── given that the tokens and strategies length are not equal
+│   │       └── it should revert
+│   └── given that the above conditions are satisfied
+│       ├── it should delete the withdrawal root from pending withdrawals
+│       ├── given that receiveAsTokens is true
+│       │   └── it should call _withdrawSharesAsTokens for each strategy to withdraw from
+│       ├── given that receiveAsTokens is false
+│       │   ├── it should loop through each strategy to withdraw from
+│       │   ├── given that the strategy is the beaconChainETHStrategy
+│       │   │   ├── it should call addShares on the eigenPodManager with the staker as the original pod owner
+│       │   │   └── given that the staker is delegated to (operator not zero address)
+│       │   │       ├── it should increase the original pod operator's shares
+│       │   │       └── it should push a stake update for the original pod operator
+│       │   ├── given that the strategy is not the beaconChainETHStrategy
+│       │   │   ├── it should call addShares on the strategyManager with the staker as the withdrawer
+│       │   │   └── it should increase the operator's shares with the staker as the withdrawer
+│       │   └── it should push an operator stake update
+│       └── it should emit a WithdrawalCompleted event
+├── when _withdrawSharesAsTokens is called (internal function) ***
+│   ├── given that the strategy is the beaconChainStrategy
+│   │   └── it should call withdrawSharesAsTokens on the eigen pod manager
+│   └── given that the strategy is not the beaconChainStrategy
+│       └── it should call withdrawSharesAsTokens on the strategy manager
+├── when migrate queued withdrawals is called
+│   └── given that the withdrawal is deleted from the strategy manager
```

Comment:
I stumbled over this phrasing. maybe just adding the word _successfully_ before `deleted` would help?

`given that the withdrawal is deleted from the strategy manager` = sounds like the withdrawal was already deleted before to me

`given that the withdrawal is successfully deleted from the strategy manager` = sounds like the deletion is more of an active change, occurring now

Reply:
agreed, changed to `given that the withdrawal succesfully deletes from the strategy manager` so its more clear its within this functions execution.

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/332/commits/d9645a9e4a9d32a3526cb5827b8ef30f3d85c5c4
Updated